cli_tools/osconfig_agent/updates_windows.go

```diff
@@ -34,16 +38,190 @@ func rebootRequired() (bool, error) {
 	return true, nil
 }
 
-func runUpdates() (bool, error) {
-	reboot, err := rebootRequired()
+func filterUpdate(classFilter, excludes map[string]struct{}, updt, updateColl *ole.IDispatch) error {
+	title, err := updt.GetProperty("Title")
 	if err != nil {
-		return false, err
+		return fmt.Errorf(`updt.GetProperty("Title"): %v`, err)
+	}
+
+	kbArticleIDsRaw, err := updt.GetProperty("KBArticleIDs")
+	if err != nil {
+		return fmt.Errorf(`updt.GetProperty("KBArticleIDs"): %v`, err)
+	}
+	kbArticleIDs := kbArticleIDsRaw.ToIDispatch()
+
+	kbArticleIDsCountRaw, err := kbArticleIDs.GetProperty("Count")
+	if err != nil {
+		return fmt.Errorf(`kbArticleIDsRaw.ToIDispatch().GetProperty("Count"): %v`, err)
+	}
+	kbArticleIDsCount, _ := kbArticleIDsCountRaw.Value().(int32)
+
+	fmt.Printf("DEBUG: filtering out KBs: %q\n", excludes)
+	for i := 0; i < int(kbArticleIDsCount); i++ {
+		kbRaw, err := kbArticleIDs.GetProperty("Item", i)
+		if err != nil {
+			return err
+		}
+		if _, ok := excludes[kbRaw.ToString()]; ok {
```

Comment:
consider separating the COM logic from the filter logic. Ideally, the filter logic would read cleanly with the COM logic abstracted away.